optimize/backtracking.go

```diff
@@ -50,12 +51,18 @@ func (b *Backtracking) Init(f, g float64, step float64) Operation {
 	if b.DecreaseFactor == 0 {
 		b.DecreaseFactor = defaultBacktrackingDecrease
 	}
+	if b.MinimumStepSize == 0 {
+		b.MinimumStepSize = minimumBacktrackingStepSize
+	}
 	if b.ContractionFactor <= 0 || b.ContractionFactor >= 1 {
 		panic("backtracking: ContractionFactor must be between 0 and 1")
 	}
 	if b.DecreaseFactor <= 0 || b.DecreaseFactor >= 1 {
 		panic("backtracking: DecreaseFactor must be between 0 and 1")
 	}
+	if b.MinimumStepSize <= 0 {
+		panic("backtracking: MinimumStepSize must be strictly positive")
```

Comment:
The equality condition is unreachable here given L54-55.
```suggestion
	if b.MinimumStepSize < 0 {
		panic("backtracking: MinimumStepSize must be positive")
```